miden-lib/asm/kernels/transaction/api.masm

```diff
@@ -178,6 +188,82 @@ export.set_account_item
     # => [R', V]
 end
 
+#! Returns VALUE located under specified KEY in map in specified account storage slot. 
+#! Panics if 
+#! - the index is out of bounds.
+#! - the requested storage slot type is not map
+#!
+#! Stack: [index, KEY, ...]
+#! Output: [VALUE]
+#!
+#! - index is the index of the item to get.
+#! - VALUE is the value of the item.
+export.get_account_map_item
+    # check if storage type is map
+    dup exec.account::get_storage_slot_type_info drop
+    # => [slot_type, index, KEY, ...]
+
+    # fails if slot_type is not 1 = map
+    push.STORAGE_MAP eq assert
+    # => [index, KEY, ...]
+    
+    # fetch the account storage item, which is ROOT of the map
+    exec.account::get_item swapw
+    # => [KEY, ROOT ...]
+
+    # fetch the VALUE located under KEY in the tree
+    exec.smt::get 
+    # => [VALUE, ROOT, ...]
+
+    # organize the stack for return
+    swapw dropw
+    # => [VALUE]
+end
+
+#! Inserts specified NEW_VALUE under specified KEY in map in specified account storage slot. 
+#! Panics if 
+#! - the index is out of bounds.
+#! - the requested storage slot type is not map
+#!
+#! Stack: [index, KEY, NEW_VALUE, ...]
+#! Output: [OLD_MAP_ROOT]
```

Comment:
**Request**: This procedure return both the old root and old value:

https://github.com/0xPolygonMiden/miden-base/blob/29ab9799ab333cb246ff6dd9d582ae600206808a/miden-lib/asm/kernels/transaction/api.masm#L264

This should be:

```
#! Output: [OLD_MAP_ROOT, OLD_MAP_VALUE]
```

Reply:
**Request**: Like `get_account_map_item` this is being called with a `sycall`, so we should make the input/output size the same. Here the output should be:

```
#! Output: [OLD_MAP_ROOT, OLD_MAP_VALUE, 0]
```